src/inmanta/deploy/persistence.py

```diff
@@ -302,16 +302,33 @@ async def dryrun_update(self, env: UUID, dryrun_result: executor.DryrunResult) -
             changes=dryrun_result.changes,
         )
 
-        await self._write_resource_action(
-            env,
-            dryrun_result.rvid,
-            const.ResourceAction.dryrun,
-            uuid.uuid4(),
-            const.ResourceState.dry,
-            dryrun_result.started,
-            dryrun_result.finished,
-            dryrun_result.messages,
-        )
+        async with data.Resource.get_connection() as connection:
+            await self._write_resource_action(
+                env,
+                dryrun_result.rvid,
+                const.ResourceAction.dryrun,
+                uuid.uuid4(),
+                const.ResourceState.dry,
+                dryrun_result.started,
+                dryrun_result.finished,
+                dryrun_result.messages,
+                connection=connection,
+            )
+
+            if dryrun_result.resource_unavailable:
+                await self._write_resource_action(
+                    env,
+                    dryrun_result.rvid,
+                    const.ResourceAction.dryrun,
+                    uuid.uuid4(),
+                    const.ResourceState.unavailable,
+                    # Small hack so that when we return the logs ordered by started time,
+                    # we don't get inconsistencies in the ordering by having 2 logs with the same started time.
+                    dryrun_result.started + datetime.timedelta(milliseconds=1),
+                    dryrun_result.finished + datetime.timedelta(milliseconds=1),
```

Comment:
I added this because the test in which I added a comment, was flaky.
It was not flaky before because we had different start times

Reply:
Why write two resource actions?

Reply:
One of them has the status as `dry` and the other as `unavailable`. It is the behaviour that was present before my changes.

Should we keep only the dry` or the `unavailable`?

Reply:
the unavailable indeed